pkg/middleware/recovery/recovery_test.go

```diff
@@ -0,0 +1,158 @@
+package recovery
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	"github.com/openfga/openfga/pkg/logger"
+)
+
+func TestPanic(t *testing.T) {
+	panicHandlerFunc := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		panic("Unexpected error!")
+	})
+
+	t.Run("With_HTTP_Panic_Recovery", func(t *testing.T) {
+		handler := HTTPPanicRecoveryHandler(panicHandlerFunc, logger.MustNewLogger("text", "info", "unix"))
+
+		req, err := http.NewRequest(http.MethodGet, "/", nil)
+		if err != nil {
+			require.NoError(t, err)
+		}
+
+		resp := httptest.NewRecorder()
+		require.NotPanics(t, func() {
+			handler.ServeHTTP(resp, req)
+		})
+
+		require.Equal(t, http.StatusInternalServerError, resp.Code)
```

Comment:
Also we should assert the body contains the internal server error `code` and `message`.

Reply:
Done